int32 rate
int32 bitrate
string endian